toolchest_client/tools/tool.py

```diff
@@ -451,6 +451,20 @@ def run(self):
 
         self._wait_for_threads_to_finish()
 
+        # Check for interrupted or failed threads
+        if ThreadStatus.INTERRUPTING in self.query_thread_statuses.values() \
+            or ThreadStatus.FAILED in self.query_thread_statuses.values() \
+                or self.terminating:
```

Comment:
Instead of a multiline if, I'd break this into a variable that makes this more readable.

E.g. `thread_failed = xxx`, then `if thread_failed or ...`